components/use-paid-mutation.js

```diff
@@ -23,39 +25,127 @@ export function usePaidMutation (mutation,
   const [getPaidAction] = useLazyQuery(GET_PAID_ACTION, {
     fetchPolicy: 'network-only'
   })
+  const [retryPaidAction] = useMutation(RETRY_PAID_ACTION)
   const waitForWalletPayment = useWalletPayment()
   const invoiceHelper = useInvoice()
   const waitForQrPayment = useQrPayment()
   const client = useApolloClient()
   // innerResult is used to store/control the result of the mutation when innerMutate runs
   const [innerResult, setInnerResult] = useState(result)
+  const { wallets: walletDefs } = useWallets()
+  const { me } = useMe()
+
+  const addPayError = (e, rest) => ({
+    ...rest,
+    payError: e,
+    error: e instanceof InvoiceCanceledError && e.actionError ? e : undefined
+  })
+
+  // walletDefs shouldn't change on rerender, so it should be safe
+  const usableWallets = walletDefs
+    .map(w => useWallet(w.def.name))
+    .filter(w => !w.def.isAvailable || w.def.isAvailable())
+    .filter(w => w.config?.enabled && canSend(w))
+
+  const waitForActionPayment = useCallback(async (invoice, { wallet, alwaysShowQROnFailure = false, persistOnNavigate = false, waitFor }, originalResponse, action) => {
+    const walletErrors = []
+    let response = originalResponse
+    let invoiceUsed = false
+
+    // ensures every invoice is used only once
+    const refreshInvoice = async () => {
+      if (invoiceUsed) {
+        const retry = await retryPaidAction({ variables: { invoiceId: parseInt(invoice.id) } })
+        response = retry.data?.retryPaidAction
+        invoice = response?.invoice
+        invoiceUsed = true
+      } else invoiceUsed = true
```

Comment:
ensure we are getting a fresh invoice everytime it is presented to a wallet or to the user.